src/metadata_service/fam_metadata_service_direct.cpp

```diff
@@ -2709,12 +2710,16 @@ Fam_Metadata_Service_Direct::Fam_Metadata_Service_Direct(bool use_meta_reg) {
     } else {
         enable_region_spanning = 0;
     }
+ 
+    metadata_path = (const char *)(config_options["metadata_path"]
```

Comment:
What if metadata_path is not provided in config file ?

Reply:
This has been already taken care in get_config_info().
 try {
            options["metadata_path"] =
                (char *)strdup((info->get_key_value("metadata_path")).c_str());
        }
        catch (Fam_InvalidOption_Exception e) {
            // If parameter is not present, then set the default.
            options["metadata_path"] = (char *)strdup("/dev/shm");
        }

Reply:
Can you change it to something like 
 options["metadata_path"] = (char *)strdup("/dev/shm/fam_metadata");